src/app/clusters/ecosystem-information-server/ecosystem-information-server.cpp

```diff
@@ -0,0 +1,368 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "ecosystem-information-server.h"
+
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace EcosystemInformation {
+namespace {
+
+constexpr size_t kDeviceNameMaxSize            = 64;
+constexpr size_t kUniqueLocationIdMaxSize      = 64;
+constexpr size_t kUniqueLocationIdsListMaxSize = 64;
+constexpr size_t kHomeLocationNameMaxSize      = 128;
+
+constexpr size_t kDeviceDirectoryMaxSize   = 256;
```

Comment:
are there any spec-required reasons for this maximum count?

If these are large devices that support STL, would it make sense to bump this up a lot of make them a configurable option?

Reply:
Yes this is a spec requirement. I don't know why there is a limit simply implementing constrains as defined.